docker/Dockerfile

```diff
@@ -1,17 +1,37 @@
 # Helper image to build renterd.
 FROM golang:1.21 AS builder
-LABEL maintainer="The Sia Foundation <info@sia.tech>"
 
+# Define arguments for build tags and to skip running go generate.
+ARG BUILD_TAGS='netgo' \
+    BUILD_RUN_GO_GENERATE='true'
+
+# Set the working directory.
 WORKDIR /renterd
 
-# Copy and build binary.
+# Copy over the go mod and sum files
+COPY go.mod go.sum ./
+
+# Download dependencies
+RUN go mod download
+
+# Copy the rest of the application code
 COPY . .
-RUN go generate ./...
-RUN go build ./cmd/renterd
+
+# Generate build metadata.
+RUN if [ "$BUILD_RUN_GO_GENERATE" = "true" ] ; then go generate ./... ; fi
+
+# Build renterd.
+RUN --mount=type=cache,target=/root/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    CGO_ENABLED=0 go build -ldflags="-s -w" -tags="${BUILD_TAGS}" ./cmd/renterd
```

Comment:
SQLite requires CGO

Reply:
ah I knew there was a reason we needed it! thanks